server/room-battle.js

```diff
@@ -631,6 +634,8 @@ class RoomBattle {
 		}
 	}
 	receive(/** @type {string[]} */ lines) {
+		this.allowTie = null;
```

Comment:
It makes more sense to call this `wantsTie`, and also, it makes more sense to make this a property of `BattlePlayer` than `Battle`

Reply:
Putting it on `BattlePlayer` also fixes all the vulns I mentioned earlier.

Reply:
In which case this would be:

```suggestion
		for (const player of this.players) player.wantsTie = false;
```